### ISSUE TITLE
(Question) About glue tasks

### ISSUE THREAD

**Issue by ZhichaoWang091732** [jiaweizzhao/GaLore#52]
Hello, thanks for your inspiring and excellent work!

I want to try full fine-tuning to compare with Galora, and I have blocked the use of Galora. However, I'm having some problems that when I try to run the glue task (i.e. mrpc) to full fine-tune roberta, I find that the eval acc doesn't change at all as the training progresses. I have ruled out a possible overfitting problem and I would like to ask the author or anyone else if there is a relevant solution.

![image](https://github.com/jiaweizzhao/GaLore/assets/46990927/ffa4e170-807e-49ae-b364-4c6a5f1c7ce6)

**Comment by jiaweizzhao** [comment]
Hi, thanks for your question. Were you using the hyperparameters and settings provided by our paper (appendix)?

**Comment by mzf666** [comment]
I have the same issue. I have checked the gradient norm and the learning rate are not zero. In the original code, once the metric is initialized, it was not refreshed and kept receiving new predictions across the whole training process. Hence, I manually reload the 'metric' using the 'metric = evaluate.load('glue', args.task_name)'.

However, after fixing this potential bug, it seems that while the eval loss of the finetuned model does changes, the accuracy and f1 score metrics remains the same.

```{python}
python run_glue_geomlrk.py \
    --model_name_or_path roberta-base \
    --task_name mrpc \
    --max_length 512 \
    --seed=1234 \
    --lora_r 8 \
    --lora_all_modules \
    --per_device_train_batch_size 16 \
    --num_train_epochs 30 \
    --learning_rate 1e-4 \
    --lr_scheduler_type linear \
    --weight_decay 0.1 
```
![image](https://github.com/user-attachments/assets/ca6df92a-4c6d-49ab-aaa3-fbddcb0f150a)


```{python}
python run_glue_geomlrk.py \
    --model_name_or_path roberta-base \
    --task_name mrpc \
    --max_length 512 \
    --seed=1234 \
    --lora_r 8 \
    --lora_all_modules \
    --per_device_train_batch_size 16 \
    --num_train_epochs 30 \
    --learning_rate 1e-2 \
    --lr_scheduler_type linear \
    --weight_decay 0.1 
```


![image](https://github.com/user-attachments/assets/bc6bcf97-c31e-4aad-a771-b7a093e0c430)

**Comment by mzf666** [comment]
> Hello, thanks for your inspiring and excellent work!
> 
> I want to try full fine-tuning to compare with Galora, and I have blocked the use of Galora. However, I'm having some problems that when I try to run the glue task (i.e. mrpc) to full fine-tune roberta, I find that the eval acc doesn't change at all as the training progresses. I have ruled out a possible overfitting problem and I would like to ask the author or anyone else if there is a relevant solution.
> 
> ![image](https://private-user-images.githubusercontent.com/46990927/339420652-ffa4e170-807e-49ae-b364-4c6a5f1c7ce6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5MjE1MDgsIm5iZiI6MTcyNDkyMTIwOCwicGF0aCI6Ii80Njk5MDkyNy8zMzk0MjA2NTItZmZhNGUxNzAtODA3ZS00OWFlLWIzNjQtNGM2YTVmMWM3Y2U2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI5VDA4NDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4MzUzYzhhOGJhNzc1ODhhOWEwNmM1ZTkyMTBjNTMxZDczNDY5NzVmODQwZDE2ZGE0MTgxOWQ3ZDlmNjM2ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mRNNsLtXCCpPiBr3fHy7xFlaS45RDfN2IA8IL389F5U)


I spent several hours in adjusting the hyperparameters. I found the adamw optimizer does work with suitable learning rate. You can try this launching command:

```{bash}
CUDA_VISIBLE_DEVICES=$cuda_idx python run_glue_geomlrk.py \
    --model_name_or_path roberta-base \
    --task_name mrpc \
    --max_length 512 \
    --seed=1234 \
    --lora_r 4 \
    --per_device_train_batch_size 16 \
    --num_train_epochs 30 \
    --learning_rate 1e-5 \
    --lr_scheduler_type linear \
    --weight_decay 0.1 \
```

This leads to the result

![image](https://github.com/user-attachments/assets/dfd99cb8-d997-4fc1-9376-2ea728d87682)


It seems that improper learning rate may drive the model to mode collapse, i.e. assigning the same logits on any input sequence. Thus, the accuracy and F1 score remains unchanged as they are doing a fixed guess.

**Comment by MaeChd** [comment]
> > Hello, thanks for your inspiring and excellent work!
> > I want to try full fine-tuning to compare with Galora, and I have blocked the use of Galora. However, I'm having some problems that when I try to run the glue task (i.e. mrpc) to full fine-tune roberta, I find that the eval acc doesn't change at all as the training progresses. I have ruled out a possible overfitting problem and I would like to ask the author or anyone else if there is a relevant solution.
> > ![image](https://private-user-images.githubusercontent.com/46990927/339420652-ffa4e170-807e-49ae-b364-4c6a5f1c7ce6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5MjE1MDgsIm5iZiI6MTcyNDkyMTIwOCwicGF0aCI6Ii80Njk5MDkyNy8zMzk0MjA2NTItZmZhNGUxNzAtODA3ZS00OWFlLWIzNjQtNGM2YTVmMWM3Y2U2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI5VDA4NDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4MzUzYzhhOGJhNzc1ODhhOWEwNmM1ZTkyMTBjNTMxZDczNDY5NzVmODQwZDE2ZGE0MTgxOWQ3ZDlmNjM2ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mRNNsLtXCCpPiBr3fHy7xFlaS45RDfN2IA8IL389F5U)
> 
> I spent several hours in adjusting the hyperparameters. I found the adamw optimizer does work with suitable learning rate. You can try this launching command:
> 
> ```
> CUDA_VISIBLE_DEVICES=$cuda_idx python run_glue_geomlrk.py \
>     --model_name_or_path roberta-base \
>     --task_name mrpc \
>     --max_length 512 \
>     --seed=1234 \
>     --lora_r 4 \
>     --per_device_train_batch_size 16 \
>     --num_train_epochs 30 \
>     --learning_rate 1e-5 \
>     --lr_scheduler_type linear \
>     --weight_decay 0.1 \
> ```
> 
> This leads to the result
> 
> ![image](https://private-user-images.githubusercontent.com/88829874/362603539-dfd99cb8-d997-4fc1-9376-2ea728d87682.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1ODIxNzcsIm5iZiI6MTczMTU4MTg3NywicGF0aCI6Ii84ODgyOTg3NC8zNjI2MDM1MzktZGZkOTljYjgtZDk5Ny00ZmMxLTkzNzYtMmVhNzI4ZDg3NjgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE0VDEwNTc1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiZjhmZWQ3MTcxMWY0MzU5OTBmZjNmZTVhYTUzNjVjZWQ1NTBlOTlmMjhkZWM2MDM1MmIzMThhNGJiMTgyNjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bt8kL0RqChQkqiZ5z709kNyZlrKGuTmQq1GIWPZEJyU)
> 
> It seems that improper learning rate may drive the model to mode collapse, i.e. assigning the same logits on any input sequence. Thus, the accuracy and F1 score remains unchanged as they are doing a fixed guess.

Hello, thank you for your answer to this question;
However, when I tried to reproduce LoRA and Roberta fine-tuning on the GLUE task, there was a phenomenon that the training loss was almost unchanged and the evaluation index was constant. I wonder if you have any attempts?
